Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,217 +14,298 @@
 //
 //******************************************************************************
 
-#include "Starboard.h"
+#import <Foundation/NSException.h>
+#import <NSOrderedSetInternal.h>
+#import <Starboard.h>
 #import <StubReturn.h>
+#import <VAListHelper.h>
 
-#include "Foundation/NSOrderedSet.h"
+// Needed to make sure that the returned array behaves like a one way proxy to an array
+@implementation _NSProxyOrderedSetArray
+- (instancetype)initWithOrderedSet:(NSOrderedSet*)orderedSet {
+    if (self = [super init]) {
+        _array = orderedSet->_arrayContainer;
+    }
+
+    return self;
+}
+
+#pragma mark NSArray
```

Comment:
Sorry was unclear. Meant to only comment on one line. The "#pragma mark NSArray" doesn't make sense to me here.